### PR TITLE
chapter 24 created

### PR DESCRIPTION
**Pull request by agustinaGallo97**
## :link: Jira Tracker reference:
* [AP - Chapter 24]  HTTP and background task - https://androidtraining.atlassian.net/browse/AT-105

## Description:
Chapter 24 created

## :camera_flash: UI preview
ezgif com-video-to-gif(18)

### CODE REVIEW

File: app/src/main/java/com/bignerdranch/android/photogallery/FlickrFetchr.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  companion object {
+    private const val TAG = "PhotoGalleryFragment"
+  }
+
+  private val flickrApi: FlickrApi
+
+  init {
+    val retrofit: Retrofit =
+      Retrofit.Builder()
+        .baseUrl("https://api.flickr.com/")
```
Comment (f7deleon) [inline app/src/main/java/com/bignerdranch/android/photogallery/FlickrFetchr.kt]: use a string resource for this

Comment (f7deleon) [reply app/src/main/java/com/bignerdranch/android/photogallery/FlickrFetchr.kt]: like in the base project

File: app/src/main/java/com/bignerdranch/android/photogallery/PhotoGalleryFragment.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      Observer { galleryItems ->
+        photoRecyclerView.adapter = PhotoAdapter(galleryItems)
+      })
+
+    setUpPhotoRecyclerView(view)
+  }
+
+  private fun setUpPhotoRecyclerView(view: View) {
+    photoRecyclerView = view.findViewById(R.id.photoRecyclerView)
+    photoRecyclerView.layoutManager = GridLayoutManager(context, 3)
```
Comment (f7deleon) [inline app/src/main/java/com/bignerdranch/android/photogallery/PhotoGalleryFragment.kt]: use constants for this 3

File: app/src/main/java/com/bignerdranch/android/photogallery/api/FlickrApi.kt [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package com.bignerdranch.android.photogallery.api
+
+import retrofit2.Call
+import retrofit2.http.GET
+
+interface FlickrApi {
+  @GET(
+    "services/rest/?method=flickr.interestingness.getList" +
```
Comment (f7deleon) [inline app/src/main/java/com/bignerdranch/android/photogallery/api/FlickrApi.kt]: use constants to build this url

File: app/src/main/res/layout/activity_photo_gallery.xml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/fragmentContainer"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".PhotoGalleryActivity"
+    />
```
Comment (f7deleon) [inline app/src/main/res/layout/activity_photo_gallery.xml]: nl

File: app/src/main/res/layout/fragment_photo_gallery.xml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.recyclerview.widget.RecyclerView
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:id="@+id/photoRecyclerView"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    />
```
Comment (f7deleon) [inline app/src/main/res/layout/fragment_photo_gallery.xml]: nl

**Review by f7deleon — APPROVED** [review]

File: app/src/main/java/com/bignerdranch/android/photogallery/FlickrFetchr.kt [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        .addConverterFactory(GsonConverterFactory.create())
+        .build()
+
+    flickrApi = retrofit.create(FlickrApi::class.java)
+  }
+
+  fun fetchPhotos(): LiveData<List<GalleryItem>> {
```
Comment (mirland) [inline app/src/main/java/com/bignerdranch/android/photogallery/FlickrFetchr.kt:35]: I'd add it in another class, maybe a source

File: app/src/main/java/com/bignerdranch/android/photogallery/GalleryItem.kt [hunk 7]
```diff
@@ -0,0 +1,9 @@
+package com.bignerdranch.android.photogallery
+
+import com.google.gson.annotations.SerializedName
+
+data class GalleryItem(
+    var title: String = "",
```
Comment (mirland) [inline app/src/main/java/com/bignerdranch/android/photogallery/GalleryItem.kt]: a rule, `data class` fields should be `val` properties

File: app/src/main/AndroidManifest.xml [hunk 8]
```diff
@@ -1,15 +1,21 @@
...
+        android:name=".PhotoGalleryApplication"
         android:allowBackup="true"
         android:icon="@mipmap/ic_launcher"
         android:label="@string/app_name"
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
-        android:theme="@style/AppTheme">
-        <activity android:name=".MainActivity">
+        android:theme="@style/AppTheme"
+        >
```
Comment (FerrariBruno) [inline app/src/main/AndroidManifest.xml:17]:
```suggestion
        >
        
```

**Review by FerrariBruno — APPROVED** [review]

**Pull request merged**